From 1dee144c272a298314d0c4832186fda019ad9d8d Mon Sep 17 00:00:00 2001
From: Sean Whitton <spwhitton@spwhitton.name>
Date: Sat, 20 Jul 2019 11:56:43 +0100
Subject: [PATCH] dgit-maint-*(7): Advice for git-debpush users

dgit-maint-native(7) doesn't need updating because git-debpush(1) is
clear that quilt mode options are required only for non-native
packages.

This commit leaves `dgit push{,-source}` as the first choice, with
git-debpush(1) given as an alternative.  We might want to change that
in the future, but for the time being this makes sense.

Signed-off-by: Sean Whitton <spwhitton@spwhitton.name>
---
 dgit-maint-debrebase.7.pod |  4 ++++
 dgit-maint-gbp.7.pod       |  3 +++
 dgit-maint-merge.7.pod     | 16 ++++++++++------
 3 files changed, 17 insertions(+), 6 deletions(-)

diff --git a/dgit-maint-debrebase.7.pod b/dgit-maint-debrebase.7.pod
index 9c9598bb..e122c20c 100644
--- a/dgit-maint-debrebase.7.pod
+++ b/dgit-maint-debrebase.7.pod
@@ -484,6 +484,10 @@ fast-forwarding from the history on B<dgit-repos>.  In such cases you
 will have to pass I<--overwrite> to dgit.  git-debrebase will normally
 tell you if this will be needed.
 
+If you want to upload with git-debpush(1), for the first upload you
+should pass the B<--quilt=linear> quilt mode option (see
+git-debpush(1)).
+
 Right before uploading, if you did not just already do so, you might
 want to have git-debrebase(1) shuffle your branch such that the Debian
 delta queue appears right at the tip of the branch you will push:
diff --git a/dgit-maint-gbp.7.pod b/dgit-maint-gbp.7.pod
index b8abd66c..1988a317 100644
--- a/dgit-maint-gbp.7.pod
+++ b/dgit-maint-gbp.7.pod
@@ -127,6 +127,9 @@ history.  (This origin commit would represent the most recent non-dgit
 upload of the package, but this should already be represented in your
 git history.)
 
+Alternatively, you can use git-debpush(1).  For the first upload you
+should pass the B<--gbp> quilt mode option (see git-debpush(1)).
+
 =head1 INCORPORATING NMUS
 
 B<dgit pull> can't yet incorporate NMUs into patches-unapplied gbp
diff --git a/dgit-maint-merge.7.pod b/dgit-maint-merge.7.pod
index 0ccd8c7e..ea2c7765 100644
--- a/dgit-maint-merge.7.pod
+++ b/dgit-maint-merge.7.pod
@@ -343,12 +343,16 @@ dgit(1).  If any command fails, dgit will provide a carefully-worded
 error message explaining what you should do.  If it's not clear, file
 a bug against dgit.  Remember to pass I<--new> for the first upload.
 
-As an alternative to B<dgit build> and friends, you can use a tool
-like gitpkg(1).  This works because like dgit, gitpkg(1) enforces that
-HEAD has exactly the contents of the source package.  gitpkg(1) is
-highly configurable, and one dgit user reports using it to produce and
-test multiple source packages, from different branches corresponding
-to each of the current Debian suites.
+If you want to upload with git-debpush(1), for the first upload you
+should pass the B<--quilt=smash> quilt mode option (see
+git-debpush(1)).
+
+As another alternative to B<dgit build> and friends, you can use a
+tool like gitpkg(1).  This works because like dgit, gitpkg(1) enforces
+that HEAD has exactly the contents of the source package.  gitpkg(1)
+is highly configurable, and one dgit user reports using it to produce
+and test multiple source packages, from different branches
+corresponding to each of the current Debian suites.
 
 If you want to skip dgit's checks while iterating on a problem with
 the package build (for example, you don't want to commit your changes
-- 
2.20.1

